lib/sync-DataSetModel.js

```diff
@@ -418,13 +467,46 @@ var self = {
         rec.hash = hash;
         recOut[i] = rec;
       }
-      var globalHash = syncUtil.generateHash(hashes);
 
-      var res = {
-        hash: globalHash,
-        records: recOut
-      };
-      return cb(null, res);
+      if (hashes.length > 0) {
+        console.log('forceSyncList - sync hashes : ' + syncUtil.generateHash(hashes));
+        console.log('forceSyncList - sync recOut : ' + syncUtil.generateHash(recOut));
+      }*/
+
+      // Start: Aiden Keating Hyper Dragon Work v4.8
+
+      var recOut = {};
+      var hashes = [];
+
+      async.forEachSeries(Object.keys(records), function(recordId, itemCallback) {
+        var record = records[recordId];
+        syncUtil.asyncGenerateHash(record, function(err, recordHash) {
+          var rec = {
+            data: record,
+            hash: recordHash
+          };
+          hashes.push(recordHash);
+          recOut[recordId] = rec;
+          itemCallback();
+        });
+      }, function(err) {
+        return syncUtil.asyncGenerateHash(hashes, function(err, globalHash) {
+          var res = {
+            hash: globalHash,
+            records: recOut
+          };
+          console.log('Async result: ' + JSON.stringify(res));
+          return cb(null, res);
+        });
+      })
+      // End: Aiden Keating Hyper Dragon Work 4.8
+
+      //var globalHash = syncUtil.generateHash(hashes);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this commented out code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ACommentedCode)